druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -1102,24 +1107,46 @@ public String getQueryString(String pagingIdentifier, int offset) {
         }
       }
       this.timeZone = timeZone;
+      this.dateFormatter = new SimpleDateFormat(TimeExtractionFunction.ISO_TIME_FORMAT,
+          Locale.ROOT);
+      if (timeZone != null) {
+        this.dateFormatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+      }
     }
 
     protected void clearFieldNameLists() {
       dimensions.clear();
       metrics.clear();
     }
 
-    @SuppressWarnings("incomplete-switch") String translate(RexNode e, boolean set) {
+    @SuppressWarnings("incomplete-switch")
+    String translate(RexNode e, boolean set, boolean formatDateString) {
```

Comment:
can we java docs about what `formatDateString` is used for?

Reply:
done.